src/test/kotlin/uk/gov/justice/digital/hmpps/visitallocationapi/integration/StartVisitAllocationByPrisonControllerTest.kt

```diff
@@ -51,23 +50,18 @@ class StartVisitAllocationByPrisonControllerTest : IntegrationTestBase() {
     val visitAllocationEventJobDto = getVisitAllocationEventJobDto(result)
     Assertions.assertThat(visitAllocationEventJobDto.totalActivePrisons).isEqualTo(2)
     verify(sqsService, times(2)).sendVisitAllocationEventToAllocationJobQueue(any(), any())
-    verify(sqsService).sendVisitAllocationEventToAllocationJobQueue(visitAllocationEventJobDto.allocationJobReference, prison1Active.prisonCode)
-    verify(sqsService).sendVisitAllocationEventToAllocationJobQueue(visitAllocationEventJobDto.allocationJobReference, prison2Active.prisonCode)
+    verify(sqsService).sendVisitAllocationEventToAllocationJobQueue(visitAllocationEventJobDto.allocationJobReference, prison1Active.prisonId)
+    verify(sqsService).sendVisitAllocationEventToAllocationJobQueue(visitAllocationEventJobDto.allocationJobReference, prison2Active.prisonId)
     verify(visitOrderAllocationJobRepository, times(1)).save(any())
     verify(visitOrderAllocationPrisonJobRepository, times(2)).save(any())
   }
 
   @Test
   fun `when no active prisons allocation job started then no sqs messages are sent`() {
     // Given
-    val prison1Inactive = VisitOrderPrison(prisonCode = prisonCode1, active = false)
-    val prison2Inactive = VisitOrderPrison(prisonCode = prisonCode2, active = false)
-    val prison3Inactive = VisitOrderPrison(prisonCode = prisonCode3, active = false)
-    entityHelper.savePrison(prison1Inactive)
-    entityHelper.savePrison(prison2Inactive)
-    entityHelper.savePrison(prison3Inactive)
 
     // When
+    prisonApiMockServer.stubGetAllServicePrisonsEnabledForDps(prisons = null, httpStatus = HttpStatus.NOT_FOUND)
     val responseSpec = startVisitAllocationByPrisonJob(webTestClient, startVisitAllocationJobRoleHttpHeaders)
 
     // Then
```

Comment:
worth adding a test to see if an INTERNAL_SERVER_ERROR error is handled gracefully